### PR TITLE
Error page

### PR DESCRIPTION
**Pull request by SenecaAnderson**
### Summary
When extraction fails, an error page will display. It explains that there was an error, displays the logged messages, and provides a button back to the Extract component.

### New Behavior
If the config file name or contents are invalid, or extraction fails for some reason, the error page will appear.

### Code Changes
A new ExtractionError component was added. Additionally, there are some minor changes to the page-text css class.

### Testing Guidance
Start the app with npm start. Click on "Extract New". There are two options to deliberately cause errors:

- Add a typo to the default config file path, or provide any invalid file path. There will be only one logged message if you do this.
- Modify the config file so that something will cause an error. If you change the last extractor file path, this will cause a long list of logged messages to appear, so you will get to see what the error page looks like with many messages.

Click "Submit". After the error occurs, the error page will appear, and you can view the logged messages. Click on "Extract New" to run the process again. If you would like to see debug-level messages, check "Log output debugging information". Note that once you set the debug flag, the flag will be set until you restart the app from the terminal, so every extraction after that will have debug-level messages.

### CODE REVIEW

**Comment by jafeltra** [comment]
Can you update the base branch of this PR to be `logger-display`? Since this branch has changes on top of that branch, it will show just the relevant differences in this PR so the right changes are reviewed ehre, and when that branch gets merged to master, the base branch of this PR will update. That will also prevent us from accidentally merging this to `master` before the `logger-display` PR is merged.

File: electron/Transport.js
Comment (jafeltra) [inline electron/Transport.js:14]: It looks like maybe an old version of the `InMemoryTransport` got added back. Is this intentional? Or can we remove this file?

File: react-app/src/components/ExtractionError.js
Comment (jafeltra) [inline react-app/src/components/ExtractionError.js:21]: Similar to my comment on the `LogList`, but I think it might be nice of the heading and directions you have here stayed at the top of the page and just the list of log messages scrolls. It might be possible to make that change within the `LogList` component, so maybe it would just come for free if you made the change in the `logger-display` branch and updated this branch. If so, then this should be set! Otherwise, it might be nice to add a similar change to this page as well.

**Review by jafeltra — CHANGES_REQUESTED** [review]
I noticed a couple small things, but overall this is looking really good! It's cool to see so many pages implemented now!

File: electron/Transport.js
Comment (SenecaAnderson) [reply electron/Transport.js:14]: The old version must have been added back in during rebasing. I've deleted it.

File: react-app/src/components/ExtractionError.js
Comment (SenecaAnderson) [reply react-app/src/components/ExtractionError.js:21]: I tried making the subtitle sticky. However, I ran into the same issue as on the logger-display branch--the LogList item covers it up when you scroll.

Comment (jafeltra) [reply react-app/src/components/ExtractionError.js:21]: It sounds like you're doing some exploration for this on the `logger-display` branch. Let's see if we can get it working as expected over on that branch, and then we can evaluate what we'll need to update here!

Comment (SenecaAnderson) [reply react-app/src/components/ExtractionError.js:21]: I created a CSS class called sticky-item in the logger-display branch. Once that branch is merged and this one is rebased, I can apply that styling to the subtitle of the error page as well.

Comment (jafeltra) [reply react-app/src/components/ExtractionError.js:21]: Since this branch is based off the `logger-display` branch, I actually think  you could rebase this branch against the updated version, and you should just get the one or two new commits from that branch on this one. If that rebase doesn't go smoothly though, we can definitely just hold off until the `logger-display` branch is merged like you said!

**Comment by mgramigna** [comment]
This looks good to me once the rebase goes through

File: react-app/src/components/PatientData.js
Comment (jafeltra) [inline react-app/src/components/PatientData.js:8]: I think we still want to have the `!props.showLogs` condition in this statement, right? I'm guessing it just got lost in a rebase.

File: react-app/src/components/ExtractionError.js
Comment (jafeltra) [reply react-app/src/components/ExtractionError.js:21]: I think it would be good to carry over the scrolling in the LogList component from the `logger-display` branch to here. When I have the app window small and a few different errors listed, the title and directions scroll up on the page when I scroll down to look at the rest of the errors. I think it would be nice to have the same interaction that the LogFile view has.

**Review by jafeltra — COMMENTED** [review]
This is looking good, I just had a few other small comments inline and one that I couldn't add to a specific line:

[JA1] - I'm not sure if some CSS property got lost in the rebasing, but the "Log File" title on the Results page isn't aligned on the left side like it is on the `logger-display` branch.

**Review by jafeltra — APPROVED** [review]
This looks good to me! Thanks for looking into that scroll, and I think it is okay as it is now since there likely won't be many errors listed on that page since it usually crashes pretty early if there are fatal errors!

**Review by mgramigna — APPROVED** [review]

**Pull request merged**
